hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/test/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapterTest.java

```diff
@@ -311,4 +320,129 @@ public void testFuturePutAreFlagged() {
     verify(metricsExporter, times(1)).incCounters(PUTS_IN_FUTURE_METRIC_KEY, 1);
     verifyNoMoreInteractions(metricsExporter);
   }
+
+  @Test
+  public void testFilterLargeCellsFlag() {
+    int numBytes = 150 * 1024 * 1024;
+    byte[] largeCellValBytes = new byte[numBytes];
+    new Random().nextBytes(largeCellValBytes);
+
+    ArrayList<Cell> walEntryCells = new ArrayList<>();
+    Cell incompatibleLargePut =
+        new KeyValue(rowKey, cf, qual, 0, KeyValue.Type.Put, largeCellValBytes);
+    Cell put = new KeyValue(rowKey, cf, qual, 0, KeyValue.Type.Put, val);
+    walEntryCells.add(put);
+    walEntryCells.add(incompatibleLargePut);
+    BigtableWALEntry walEntry =
+        new BigtableWALEntry(System.currentTimeMillis(), walEntryCells, tableName);
+
+    Assert.assertEquals(
+        conf.getBoolean(FILTER_LARGE_CELLS_KEY, DEFAULT_ENABLED_FILTER_LARGE_CELLS),
+        DEFAULT_ENABLED_FILTER_LARGE_CELLS);
+    Assert.assertEquals(
```

Comment:
Why is the cell not getting filter? The filter large cells is enabled by default, so I don't understand how incompatibleLargePut is returned.

Reply:
filtering is always disabled by default unless the end user configures it. this test is a negative test, to ensure we do just that.